File: TextosJapones/src/main/java/org/jisho/textosJapones/components/listener/VinculoListener.java
package org.jisho.textosJapones.components.listener;

import javafx.scene.Node;
import org.jisho.textosJapones.model.entities.VinculoPagina;
import org.jisho.textosJapones.model.enums.Pagina;

public interface VinculoListener {
	Boolean onDuploClique(Node root, VinculoPagina vinculo, Pagina origem);

	void onDrop(Pagina origem, VinculoPagina vinculoOrigem, Pagina destino, VinculoPagina vinculoDestino);

	void onDragStart();

	void onDragEnd();
}

File: ProcessaTexto/src/main/kotlin/br/com/fenix/processatexto/components/listener/VinculoListener.kt
package br.com.fenix.processatexto.components.listener

import br.com.fenix.processatexto.model.entities.processatexto.VinculoPagina
import br.com.fenix.processatexto.model.enums.Pagina
import javafx.scene.Node

interface VinculoListener {
    fun onDuploClique(root: Node, vinculo: VinculoPagina, origem: Pagina): Boolean
    fun onDrop(origem: Pagina, vinculoOrigem: VinculoPagina, destino: Pagina, vinculoDestino: VinculoPagina)
    fun onDragStart()
    fun onDragEnd()
}